### PR TITLE
[v7r0] Reduced (removed, in fact) interactions of Optimizers with JobParameters

### PR DESCRIPTION
**Pull request by fstagni**
And just interacting with OptimizerParameters

BEGINRELEASENOTES

*WMS
CHANGE: Reduced (removed, in fact) interactions of Optimizers with JobParameters, using only OptimizerParameters

ENDRELEASENOTES

### CODE REVIEW

File: WorkloadManagementSystem/Client/JobState/CachedJobState.py [hunk 1]
```diff
@@ -155,107 +158,108 @@ def deserialize( stub ):
...
+        if len(cKey) != len(data):
+          gLogger.warn(
+              "CachedJobState.__memorize( %s, %s = %s ) doesn't receive the same amount of values as keys" %
+              (cKey, functor, data))
           return data
-        for i in range( len( cKey ) ):
-          self.__cache[ cKey[ i ] ] = data[i]
-      #Prepare result
-      return S_OK( tuple( [ self.__cache[ cK ]  for cK in cKey ] ) )
+        for i in range(len(cKey)):
```
Comment (andresailer) [inline WorkloadManagementSystem/Client/JobState/CachedJobState.py]: `enumerate`

File: WorkloadManagementSystem/Executor/test/Test_Executor.py [hunk 2]
```diff
@@ -1,100 +1,28 @@
...
-# Test Suite run
-#############################################################################
-
-if __name__ == '__main__':
-  suite = unittest.defaultTestLoader.loadTestsFromTestCase( ExecutorTestCase )
-  suite.addTest( unittest.defaultTestLoader.loadTestsFromTestCase( JobSchedulingSuccess ) )
-  testResult = unittest.TextTestRunner( verbosity = 2 ).run( suite )
-
-# EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#EOF#
+@pytest.mark.parametrize("sites, banned, expected", [
```
Comment (chaen) [inline WorkloadManagementSystem/Executor/test/Test_Executor.py]: yes !! :)

File: WorkloadManagementSystem/Executor/Base/OptimizerExecutor.py [hunk 3]
```diff
@@ -1,182 +1,183 @@
...
 """
 
+
+__RCSID__ = "$Id$"
+
 import threading
-#Because eval(valenc) might require it
-import datetime
-from DIRAC import S_OK, S_ERROR, gLogger
+# Because eval(valenc) might require it
```
Comment (andresailer) [inline WorkloadManagementSystem/Executor/Base/OptimizerExecutor.py]: this comment belongs to `import datetime`, probably means `import datetime` should be kept

**Comment by atsareg** [comment]
I guess this is done for optimization purpose. Note that optimizer parameters unlike job parameters are not visible through the web interface, which should be probably added.

**Comment by fstagni** [comment]
Yes, I thought about it. This requires a little dev in WebApp.

**Comment by fstagni** [comment]
I have added a `export_getJobOptParameters(jobID):` to `JobMonitoringHandler`. Now it needs to exposed to the web, I let @zmathe take care (it can go directly with WebApp4.0)

**Pull request merged**